### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by Voles**
1. Update @octokit/rest to version v18.x (as proposed in https://github.com/semantic-release/github/pull/250#issuecomment-697413090)
The breaking changes in v18.x is the removal of previously deprecated APIs. See their [changelog](https://github.com/octokit/rest.js/releases/tag/v18.0.0).

2. Upgrade nock to version [v13.x](https://github.com/nock/nock/releases/tag/v13.0.0)
I had to add the `prerelease: true` property to an integration test. The test was failing without it after this upgrade. I'm not sure why, maybe someone can double check this?

3. Upgrade all other dependencies in the `dependencies` section. All tests kept passing.

### CODE REVIEW

File: test/integration.test.js [hunk 1]
```diff
@@ -209,6 +209,7 @@ test.serial('Publish a release with release information in assets', async (t) =>
       target_commitish: options.branch,
       name: nextRelease.gitTag,
       body: nextRelease.notes,
+      prerelease: true,
```
Comment (gr2m) [inline test/integration.test.js:212]: hmm that does not look right, I'll have to investigate. Does that only occur with the upgrade of nock 12 to 13?

Comment (Voles) [reply test/integration.test.js:212]: I've did some more research. On nock v13.0.2, everything still works as expected.
Our tests start failing from nock@13.0.3 onwards.

See the release notes: https://github.com/nock/nock/releases/tag/v13.0.3

The output below is what I get from running the tests with nock@13.0.3

You can see that the body expects `prerelease: true`, but we have not added that property to our test. https://github.com/semantic-release/github/blob/master/test/integration.test.js#L207

```
 integration › Publish a release with release information in assets

  Rejected promise returned by test. Reason:

  HttpError (RequestError) {
    attemptNumber: 4,
    headers: {},
    request: {
      body: '{"tag_name":"v1.0.0","name":"v1.0.0","body":"Test release note body","prerelease":true,"draft":true}',
      headers: {
        accept: 'application/vnd.github.v3+json',
        authorization: 'token [REDACTED]',
        'content-type': 'application/json; charset=utf-8',
        'user-agent': 'octokit-rest.js/17.11.2 octokit-core.js/2.5.4 Node.js/12.18.3 (macOS Catalina; x64)',
      },
      method: 'POST',
      request: {
        agent: undefined,
        hook: Function bound bound register {},
      },
      url: 'https://api.github.com/repos/test_user/test_repo/releases',
    },
    retriesLeft: 0,
    status: 500,
    message: `request to https://api.github.com/repos/test_user/test_repo/releases failed, reason: Nock: No match for request {␊
      "method": "POST",␊
      "url": "https://api.github.com/repos/test_user/test_repo/releases",␊
      "headers": {␊
        "accept": [␊
          "application/vnd.github.v3+json"␊
        ],␊
        "user-agent": [␊
          "octokit-rest.js/17.11.2 octokit-core.js/2.5.4 Node.js/12.18.3 (macOS Catalina; x64)"␊
        ],␊
        "authorization": [␊
          "token github_token"␊
        ],␊
        "content-type": [␊
          "application/json; charset=utf-8"␊
        ],␊
        "content-length": [␊
          "100"␊
        ],␊
        "accept-encoding": [␊
          "gzip,deflate"␊
        ],␊
        "connection": [␊
          "close"␊
        ]␊
      },␊
      "body": "{\\"tag_name\\":\\"v1.0.0\\",\\"name\\":\\"v1.0.0\\",\\"body\\":\\"Test release note body\\",\\"prerelease\\":true,\\"draft\\":true}"␊
    }`,
  }

  › }
  › node_modules/@octokit/request/dist-src/fetch-wrapper.js:88:15

  ─

  1 test failed
```

Comment (gr2m) [reply test/integration.test.js:212]: I'm sorry I didn't get back to you yet :( I did update `@ocotkit/rest` to latest yet, and we will get a new release via https://github.com/semantic-release/github/pull/298. Plus we are in the process of enabling renovate to get automated dependency updates again.

**Comment by gr2m** [comment]
that should be resolved  now

**Pull request closed without merging**